### PR TITLE
Add small utility function to check for gpmaint across groups

### PR DESCRIPTION
**Pull request by joakim-hove**

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/Schedule/ScheduleState.cpp [hunk 1]
```diff
@@ -376,4 +376,15 @@ bool ScheduleState::rst_file(const RSTConfig&  rst,
...
 
 
+bool ScheduleState::has_gpmaint() const {
+    for (const auto& [_, group_ptr] : this->groups) {
+        (void)_;
+        const auto& gpmaint = group_ptr->gpmaint();
+        if (gpmaint.has_value())
+            return true;
+    }
+    return false;
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/Schedule/ScheduleState.cpp:386]: Better:
```C++
return std::any_of(this->groups.begin(), this->groups.end(),
    [](const auto& group)
{
    return group.second->gpmaint().has_value();
});
```

Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/ScheduleState.cpp]: OK - updated.

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**